WrightTools/kit/_utilities.py

```diff
@@ -0,0 +1,139 @@
+                        tup[1].im_func.func_name[-2:] == '__')
+            if internal and exclude_internal:
+                include = False
+            else:
+                include = (bound_to == the_class and not instance_only) or (
+                    bound_to is None and not class_only)
+        else:
+            include = False
+        return include
+
+    # filter to return results according to internal function and arguments
+    tups = filter(acceptMethod, inspect.getmembers(the_class))
+    return [tup[0] for tup in tups]
+
+
+def string2identifier(s):
+    """Turn a string into a valid python identifier.
+
+    Parameters
+    ----------
+    s : string
+        string to convert
+
+    Returns
+    -------
+    str
+        valid python identifier.
+    """
+    # https://docs.python.org/3/reference/lexical_analysis.html#identifiers
+    if s[0] not in string.ascii_letters:
+        s = '_' + s
+    valids = string.ascii_letters + string.digits + '_'
+    out = ''
+    for i, char in enumerate(s):
+        if char in valids:
+            out += char
+        else:
+            out += '_'
+    return out
+
+
+# --- classes -------------------------------------------------------------------------------------
+
+
+class Suppress(object):
+    """Context manager for doing a "deep suppression" of stdout and stderr in Python.
+
+    i.e. will suppress all print, even if the print originates in a
+    compiled C/Fortran sub-function.
+
+    This will not suppress raised exceptions, since exceptions are printed
+    to stderr just before a script exits, and after the context manager has
+    exited (at least, I think that is why it lets exceptions through).
+
+    `Source`__
+
+    __ http://stackoverflow.com/questions/11130156/suppress-stdout-stderr-print-from-python-functions
+
+
+    >>> with WrightTools.kit.Supress():
+    ...     rogue_function()
+
+    """
+
+    def __init__(self):
+        """init."""
+        # Open a pair of null files
+        self.null_fds = [os.open(os.devnull, os.O_RDWR) for x in range(2)]
+        # Save the actual stdout (1) and stderr (2) file descriptors.
+        self.save_fds = (os.dup(1), os.dup(2))
+
+    def __enter__(self):
+        """enter."""
+        # Assign the null pointers to stdout and stderr.
+        os.dup2(self.null_fds[0], 1)
+        os.dup2(self.null_fds[1], 2)
+
+    def __exit__(self, *_):
+        """exit."""
+        # Re-assign the real stdout/stderr back to (1) and (2)
+        os.dup2(self.save_fds[0], 1)
+        os.dup2(self.save_fds[1], 2)
+        # Close the null files
+        os.close(self.null_fds[0])
+        os.close(self.null_fds[1])
+
+
+class Timer:
+    """Context manager for timing code.
+
+    >>> with Timer():
+    ...     your_code()
+    """
+
+    def __init__(self, verbose=True):
+        """init."""
+        self.verbose = verbose
+
+    def __enter__(self, progress=None):
+        """enter."""
+        self.start = clock()
```

Comment:
import for clock

Reply:
Note that as of Python 3.3, clock is deprecated in favor of `perf_counter` if you want wall-time (including sleep) or `process_time` if you want the cpu time. In fact, for the purpose that it is used here, there exists the [`timeit`](https://docs.python.org/3/library/timeit.html) module in the standard library, which may accomplish what we desire (though not as a context manager)